game-core/src/test/java/games/strategy/triplea/delegate/WW2V3Year41Test.java

```diff
@@ -1368,7 +1366,7 @@ void testAirCanLandWithAlliedFighters() {
     addTo(sz40, carrier(gameData).create(1, germans));
     addTo(sz40, fighter(gameData).create(1, italians(gameData)));
     addTo(madagascar, fighter(gameData).create(2, germans));
-    final Route route = gameData.getMap().getRoute(madagascar, sz40);
+    final Route route = gameData.getMap().getRoute(madagascar, sz40, Matches.always());
```

Comment:
What was the motivation for having an 'always match' parameter? Seems like this is a case for method overloading and having a simplified API that only takes the two parameters, let the class handle under-the-covers that the third parameter is defaulted to a tautology.

Reply:
Ah, I actually figured you'd enjoy this change. The reason is that the method was mostly only used in test code and its generally a bad method to use. You almost always should have some condition for the route you want to find based on the units that you expect to move. Most of these tests really should have a condition and are technically passing either because it isn't checking the route closely or the only route between them happens to meet the requirements. In this example, it should really have a condition that air units can actually move into the territories along the route.